cmd/okctl/adduser.go

```diff
@@ -0,0 +1,117 @@
+package main
+
+import (
+	"regexp"
+
+	"github.com/go-ozzo/ozzo-validation/v4/is"
+
+	validation "github.com/go-ozzo/ozzo-validation/v4"
+	"github.com/oslokommune/okctl/pkg/api"
+	"github.com/oslokommune/okctl/pkg/okctl"
+	"github.com/oslokommune/okctl/pkg/spinner"
+	"github.com/spf13/cobra"
+)
+
+const (
+	addUserArgs = 2
+)
+
+// AddUserOpts options for the adduser command
+type AddUserOpts struct {
+	Environment    string
+	AWSAccountID   string
+	RepositoryName string
+	Region         string
+	ClusterName    string
+	UserPoolID     string
+	UserEmail      string
+}
+
+// Validate the inputs
+func (o *AddUserOpts) Validate() error {
+	return validation.ValidateStruct(o,
+		validation.Field(&o.Environment, validation.Required, validation.Match(regexp.MustCompile("^[a-zA-Z]{3,64}$")).Error("must consist of 3-64 characters (a-z, A-Z)")),
+		validation.Field(&o.AWSAccountID, validation.Required, validation.Match(regexp.MustCompile("^[0-9]{12}$")).Error("must consist of 12 digits")),
+		validation.Field(&o.RepositoryName, validation.Required),
+		validation.Field(&o.Region, validation.Required),
+		validation.Field(&o.ClusterName, validation.Required),
+		validation.Field(&o.UserPoolID, validation.Required),
+		validation.Field(&o.UserEmail, validation.Required, is.Email),
+	)
+}
+
+// nolint: funlen
+func buildAddUserCommand(o *okctl.Okctl) *cobra.Command {
+	opts := &AddUserOpts{}
+	cmd := &cobra.Command{
+		Use:   "adduser [env] [email]",
+		Short: "Add a user to identitypool",
```

Comment:
I don't think the word "identitypool" means anything to people, except those familiar with Cognito already.

Is it possible to come up with more understandable wording here?

Not sure if any of these are correct, but just to attempt making a suggestion:
"Add a user to the cluster"
"Give a user access to the cluster's services"

Reply:
Ok - will clarify a bit :)